lib/src/sai_redis_generic_get_stats.cpp

```diff
@@ -0,0 +1,192 @@
+#include "sai_redis.h"
+#include "meta/saiserialize.h"
+
+sai_status_t internal_redis_get_stats_process(
+        _In_ sai_object_type_t object_type,
+        _In_ uint32_t count,
+        _Out_ uint64_t *counter_list,
+        _In_ swss::KeyOpFieldsValuesTuple &kco)
+{
+    SWSS_LOG_ENTER();
+
+    // key:         sai_status
+    // field:       stat_id
+    // value:       stat_value
+
+    const std::string &key = kfvKey(kco);
+    const std::vector<swss::FieldValueTuple> &values = kfvFieldsValues(kco);
+
+    std::string str_sai_status = key;
+
+    sai_status_t status;
+
+    sai_deserialize_status(str_sai_status, status);
+
+    if (status == SAI_STATUS_SUCCESS)
+    {
+        uint32_t i = 0;
+        for (const auto &v : values)
+        {
+            uint64_t value = 0;
+
+            value = stoull(fvValue(v));
+            counter_list[i] = value;
+            i++;
+        }
+    }
+
+    return status;
+}
+
+// Another way to implement this would be to pass sai_serialize_ ## type ## stat as a parameter,
+// but the decision point of what function to pass is up in sai_redis_internal.h, which would make us
+// pass a callback through about five function calls, so for the sake of simplicity here's a macro that
+// generates overloaded functions
+#define DECLARE_SERIALIZE_COUNTER_ID_LIST(type)                                                               \
```

Comment:
It doesn't make a significant difference. I'll still need to define type traits for every type, and we will have a duplication in api because all other code uses old serialize/deserialize functions etc. and make metadata api inconsistent with other code. We already have similar defines in all sairedis_internal.h so I don't really know what's the benefit of doing such changes.